src/runtime/propensity.js

```diff
@@ -15,7 +15,23 @@
  */
 import * as PropensityApi from '../api/propensity-api';
 import {PropensityServer} from './propensity-server';
-import {isObject} from '../utils/types';
+import {isObject,isEnumValue} from '../utils/types';
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import {Event} from '../api/propensity-api';
+
+/** @private @const {!Object<string,AnalyticsEvent>} */
+const PropensityEventToAnalyticsEvent = {
+  [Event.IMPRESSION_PAYWALL]: AnalyticsEvent.IMPRESSION_PAYWALL,
```

Comment:
why square brackets?

Reply:
If you take out the square brackets the syntax doesn't compile.  I wasn't sure how to actually make the keys defined from an enum value until I saw it done with the [...] in another file.